src/daft-core/src/array/ops/approx_count_distinct.rs

```diff
@@ -0,0 +1,40 @@
+use arrow2::array::PrimitiveArray;
+use common_error::DaftResult;
+
+use crate::array::ops::as_arrow::AsArrow;
+use crate::array::ops::DaftApproxCountDistinctAggable;
+use crate::array::DataArray;
+use crate::datatypes::{Field, UInt64Array};
+use crate::utils::hyperloglog::HyperLogLog;
+use crate::DataType;
+
+impl DaftApproxCountDistinctAggable for UInt64Array {
```

Comment:
For these, the data already fits in memory and we return the approx count. Instead of using HLL, we can just just put the hashes into a HashSet with an identity hasher.

Reply:
Would this be more expensive? In the original case, we're just performing a mask, a few bitshifts, an increment, and an if-check (finding the max). With the hashset implementation, the hashset will need to internally perform another hash.

Will this cause a decrease in performance?

Reply:
Btw, updated the implementation to use a hashset. If we decide that the HLL implementation will also work fine for the 1-partition case, I can always the commit back.

Reply:
You can use an identity hasher which will allow you to use the Hash you calculated as the hash that goes into the hashset. Then when a collision happens it would only compare hashes.

https://github.com/Eventual-Inc/Daft/blob/2e8841dbd8a1a31882c8dc1752f158c240242a85/src/daft-table/src/ops/hash.rs#L46

Reply:
Updated!